platforms/platforms_windows.go

```diff
@@ -18,8 +18,24 @@ package platforms
 
 import (
 	specs "github.com/opencontainers/image-spec/specs-go/v1"
+	"golang.org/x/sys/windows/registry"
 )
 
+func isClientOS() bool {
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\Windows NT\CurrentVersion`, registry.QUERY_VALUE)
+	if err != nil {
+		return false
+	}
+	defer k.Close()
+
+	installationType, _, err := k.GetStringValue("InstallationType")
+	if err != nil {
+		return false
+	}
+
+	return installationType == "Client"
+}
```

Comment:
@thaJeztah Thinking on your comment, this kind of feels like it could be a helper in hcsshim/osversion also..

Reply:
@kevpar Curious on your thoughts. This regkey approach was recommended at least as an alternative to [GetProductInfo](https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getproductinfo?redirectedfrom=MSDN): https://learn.microsoft.com/en-us/previous-versions/windows/desktop/legacy/ee391629(v=vs.85)